ffi/dotnet/Devolutions.IronRdp/src/Framed.cs

```diff
@@ -0,0 +1,80 @@
+using Devolutions.IronRdp;
+
+public class Framed<S> where S : Stream
+{
+    private S stream;
+    private List<byte> buffer;
+
+    public Framed(S stream)
+    {
+        this.stream = stream;
+        this.buffer = new List<byte>();
+    }
+
+    public (S, List<byte>) GetInner()
+    {
+        return (this.stream, this.buffer);
+    }
+
+    public byte[] Peek()
+    {
+        return this.buffer.ToArray();
```

Comment:
issue: This is copying all the elements into a new array. Return `Span<T>`.

https://learn.microsoft.com/en-us/archive/msdn-magazine/2018/january/csharp-all-about-span-exploring-a-new-net-mainstay

Reply:
updated